File: api/Model/Preset.cs
using System;
using System.Collections.Generic;

public class Preset
{
  public string Alias { get; set; }
  public string AliasCategory { get; set; }
  public string Tooltip { get; set; }
  public string ProjectName { get; set; }
  public string TemplateOrigin { get; set; }
  public string OutputFolderPath { get; set; }
  public string FileKeywordTypesExtensions { get; set; }
  public List<Keyword> Keywords { get; set; }
  public bool AddToSourceControl { get; set; }
  public AutomationUpdates AutomationUpdates { get; set; }

  public Preset()
  {
    Keywords = new List<Keyword>();
    AutomationUpdates = new AutomationUpdates();
  }
}

File: client/src/app/create-template/preset.ts
export interface Preset {
  alias: string;
  aliasCategory: string;
  tooltip: string;
  projectName: string;
  templateOrigin: string;
  outputFolderPath: string;
  fileKeywordTypesExtensions: string;
  keywords: Keyword[];
  addToSourceControl: boolean;
  useAutomationUpdates: boolean;
}

export interface Keyword {
  keywordName: string;
  replacement: string;
  keywordType: string;
  showInGenerate: boolean;
}
